base/abstractdict.jl

```diff
@@ -579,7 +579,9 @@ end
 
 function setindex!(d::IdDict{K,V}, @nospecialize(val), @nospecialize(key)) where {K, V}
     !isa(key, K) && throw(ArgumentError("$(limitrepr(key)) is not a valid key for type $K"))
-    val = convert(V, val)
+    if !(val isa V) && V !== Any
```

Comment:
As `val isa Any`, if `!(val isa V)`, then `V !== Any`.

Reply:
Maybe add a comment to explain that a no-op convert still takes time in this case?

Reply:
ops, I meant `||` so it seems this is not needed.